department-of-veterans-affairs/vets-website
# Add moment date validation

## Summary

- Adds date validation to ensure the date is real, on top of if the date is syntactically correct.

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/68140

## Testing done

- Added a spec
- Confirmed the date will not be saved when invalid.

Steps
- http://localhost:3001/view-change-dependents/add-remove-form-21-686c/introduction (228 worked for me)
- update api/app/serializers/valid_va_file_number_serializer.rb#valid_va_file_number to return true
- check the box for a 18-23 yr old & minor under 18.
- Proceed through form submission. On review page set "official start date" to 10/31/2023. Then set month to September. 
- Ensure validation flags date as invalid.
- 
![image](https://github.com/department-of-veterans-affairs/vets-website/assets/133443504/12d38c0e-caf0-4328-8b85-4f6b2277f72d)

## Screenshots

Spec / validation change so no SS besides the above for testing clarity.

## What areas of the site does it impact?

form 21-674

## Acceptance criteria

### Quality Assurance & Testing

- [ x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x ] Linting warnings have been addressed

### Error Handling

- [ x] Browser console contains no warnings or errors.

### Authentication

- [ x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

No requested Feedback
